### DISCUSSION TITLE
`GraphStore`: how to split the data `RandomLinkSplit`

### DISCUSSION THREAD

**Discussion by denadai2** [pyg-team/pytorch_geometric#8749] [Q&A] [answered]
Dear PyG community,

I am using the `GraphStore` and `FeatureStore` to scale up GNNs. Is there an easy way to split data with something like `RandomLinkSplit`? I do not understand where I can specify `edge_label_index` or something like that.

Moreover, are there plans for edge features?

Thaaaank you

**Comment by rusty1s** [comment, marked as answer]
Transforms currently do not know about the concept of `GraphStore`/`FeatureStore`, and I am afraid you would need to take care of splitting the data by yourself for now. `edge_label_index` would be then used as usual, i.e., as input argument to the `LinkLoader`.

Plans for adding edge features are in the making. `FeatureStore` can handle edge-level features by now, but they are not yet properly filtered during feature fetching.

**Comment by denadai2** [comment]
Ohh thanks! But then, where should I store edge_label_index? In the feature store?Thank you!Sent from my iPhoneOn 11 Jan 2024, at 09.12, Matthias Fey ***@***.***> wrote:﻿
Transforms currently do not know about the concept of GraphStore/FeatureStore, and I am afraid you would need to take care of splitting the data by yourself for now. edge_label_index would be then used as usual, i.e., as input argument to the LinkLoader.
Plans for adding edge features are in the making. FeatureStore can handle edge-level features by now, but they are not yet properly filtered during feature fetching.

**Reply by rusty1s** [reply]
`edge_label_index` wouldn't be stored in the `GraphStore` IMO, it should be just an in-memory tensor.

**Comment by denadai2** [comment]
thanks!
